contracts/RouterStrategy.sol

```diff
@@ -0,0 +1,232 @@
+            // debtOutstanding 100, profit 50, _amountFreed 100, _loss 50
+            // loss should be 0, (50-50)
+            // profit should endup in 0
+            _loss = _loss.sub(_profit);
+            _profit = 0;
+        } else {
+            // Example:
+            // debtOutstanding 100, profit 50, _amountFreed 140, _loss 10
+            // _profit should be 40, (50 profit - 10 loss)
+            // loss should end up in be 0
+            _profit = _profit.sub(_loss);
+            _loss = 0;
+        }
+    }
+
+    function adjustPosition(uint256 _debtOutstanding) internal override {
+        if (emergencyExit) {
+            return;
+        }
+
+        uint256 balance = balanceOfWant();
+        if (balance > 0) {
+            _checkAllowance(address(yVault), address(want), balance);
+            yVault.deposit();
+        }
+    }
+
+    function liquidatePosition(uint256 _amountNeeded)
+        internal
+        override
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        uint256 balance = balanceOfWant();
+        if (balance >= _amountNeeded) {
+            return (_amountNeeded, 0);
+        }
+
+        uint256 toWithdraw = _amountNeeded.sub(balance);
+        _withdrawFromYVault(toWithdraw);
+
+        uint256 totalAssets = balanceOfWant();
+        if (_amountNeeded > totalAssets) {
+            _liquidatedAmount = totalAssets;
+            _loss = _amountNeeded.sub(totalAssets);
+        } else {
+            _liquidatedAmount = _amountNeeded;
+        }
+    }
+
+    function _withdrawFromYVault(uint256 _amount) internal {
+        if (_amount == 0) {
+            return;
+        }
+
+        uint256 _balanceOfYShares = yVault.balanceOf(address(this));
+        uint256 sharesToWithdraw =
+            Math.min(_investmentTokenToYShares(_amount), _balanceOfYShares);
+
+        if (sharesToWithdraw == 0) {
+            return;
+        }
+
+        yVault.withdraw(sharesToWithdraw, address(this), maxLoss);
+    }
+
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 _amountFreed)
+    {
+        return
+            yVault.withdraw(
+                yVault.balanceOf(address(this)),
+                address(this),
+                maxLoss
+            );
+    }
+
+    function prepareMigration(address _newStrategy) internal override {
+        IERC20(yVault).safeTransfer(
+            _newStrategy,
+            IERC20(yVault).balanceOf(address(this))
+        );
+    }
+
+    function protectedTokens()
+        internal
+        view
+        override
+        returns (address[] memory ret)
+    {
+        ret = new address[](1);
+        ret[0] = address(yVault);
+        return ret;
+    }
+
+    function ethToWant(uint256 _amtInWei)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        return _amtInWei;
+    }
+
+    function setMaxLoss(uint256 _maxLoss) public onlyEmergencyAuthorized {
```

Comment:
Change it to a local modifier, onlyManagement() (we should add it to base Strategy)

```
   modifier onlyManagement() {
        require(
            msg.sender == governance() || msg.sender == vault.management(),
            "!authorized"
        );
        _;
    }
    
    ```

Reply:
i added an issue for it but in the meantime we can do customn here

https://github.com/yearn/yearn-vaults/issues/413

Reply:
done!